src/main/java/seedu/address/logic/commands/ExportCommand.java

```diff
@@ -48,18 +63,60 @@ public void updateCsvFilePath(String filePath) {
     }
 
     /**
-     * Creates the directory for the CSV file if it does not exist.
+     * Gets the current JSON file path in which the filtered lists of persons will be written to.
      *
-     * @param csvFile The CSV file for which the directory needs to be created.
-     * @throws CommandException If directory fails to be created.
+     * @return The file path that the filtered persons list is being written to.
      */
-    public void createCsvDirectory(File csvFile) throws CommandException {
-        File csvParentDirectory = csvFile.getParentFile();
-        if (!csvParentDirectory.exists()) {
-            boolean isCreated = csvParentDirectory.mkdir();
-            if (!isCreated) {
-                throw new CommandException("Could not create directory for CSV file.");
-            }
+    public Path getFilteredJsonFilePath() {
+        return this.filteredJsonFilePath;
+    }
+
+    /**
+     * Updates the filtered JSON file path in which the filtered lists of persons will be written to.
+     *
+     * @param filePath The JSON file path to be written to.
+     */
+    public void updateFilteredJsonFilePath(Path filePath) {
+        this.filteredJsonFilePath = filePath;
+    }
+
+    /**
+     * Checks if the list is empty and throws an exception if the list is empty.
+     *
+     * @param personList The list in which to check if it is empty.
+     * @throws CommandException If the list is empty.
+     */
+    public void checkForEmptyList(ObservableList<Person> personList) throws CommandException {
+        if (personList.isEmpty()) {
+            throw new CommandException(MESSAGE_NOTHING_TO_EXPORT_FAILURE);
+        }
+    }
+
+    /**
+     * Adds all persons in ObservableList to AddressBook.
+     *
+     * @param addressBook The address book where persons should be added into.
+     * @param personList List containing persons to be added into address book.
+     */
+    public void addToAddressBook(AddressBook addressBook, ObservableList<Person> personList) {
+        for (Person person : personList) {
+            addressBook.addPerson(person);
+        }
+    }
+
+    /**
+     * Writes the information in the address book to a JSON file.
+     *
+     * @param jsonAddressBookStorage The JSON file to write information into.
+     * @param addressBook The address book from which to read the information from.
+     * @throws CommandException If information cannot be written into the JSON file.
+     */
+    public void writeToJsonFile(JsonAddressBookStorage jsonAddressBookStorage, AddressBook addressBook)
```

Comment:
Nice that you have used the classes already implemented in the storage package. This saves us a lot of time